docs/source/references/modules.rst

```diff
@@ -12,6 +12,25 @@ metrics.
 .. autoclass:: torchmetrics.Metric
     :noindex:
 
+.. note::
+
+    Metric states are being synchornized by default in a distributed setting when using :meth:`~torchmetrics.Metric.state_dict`.
+    This behaviour can be blocked in the following ways.
+    We strongly recommend calling :meth:`~torchmetrics.Metric.state_dict` on all processes or this can cause a deadlock.
+
+.. doctest::
+
+    >>> from torchmetrics import Metric
+    >>> metric = Metric(should_sync_state_dict=False)
+    >>> metric.state_dict()
+
+.. doctest::
```

Comment:
maybe explain the difference here or when to use which way?

Reply:
Done !